vendor-extra/LiberoPageBundle/src/EventListener/HomepageContentListListener.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\LiberoPageBundle\EventListener;
+
+use FluentDOM;
+use FluentDOM\DOM\Document;
+use GuzzleHttp\ClientInterface;
+use Libero\LiberoPageBundle\Event\CreatePagePartEvent;
+use Libero\LiberoPageBundle\Event\LoadPageEvent;
+use Libero\ViewsBundle\Views\ViewConverter;
+use Psr\Http\Message\ResponseInterface;
+use function var_dump;
+
+final class HomepageContentListListener
+{
+    private $client;
+    private $converter;
+
+    public function __construct(ClientInterface $client, ViewConverter $converter)
+    {
+        $this->client = $client;
+        $this->converter = $converter;
+    }
+
+    public function onLoadPage(LoadPageEvent $event) : void
+    {
+        $request = $event->getRequest();
+        $page = $request->attributes->get('libero_page');
+
+        if ('homepage' !== $page['type']) {
+            return;
+        }
+
+        $list = $this->client
+            ->requestAsync(
+                'GET',
+                "{$page['content_service']}/items",
+                [
+                    'headers' => ['Accept' => 'application/xml'],
+                    'http_errors' => true,
+                ]
+            )
+            ->then(
+                function (ResponseInterface $response) use ($event) : Document {
```

Comment:
since the `Document` returned here is a Libero wrapper type with no variation, can it be made stricter? Should the `Document` coming from the library be wrapped into a class?

Reply:
The `addDocument()` likely clashes with wrapping

Reply:
Ah it doesn't because everything is in a `PromiseInterface` anyway